bin/extract_metrics/extract_metrics.py

```diff
@@ -0,0 +1,339 @@
+from zipfile import ZipFile
+
+import asyncpg
+import boto3
+import polars as pl
+from pulse3D import metrics
+from pulse3D.data_loader.utils import get_metadata_cls
+from pulse3D.data_loader import InstrumentTypes
+from pulse3D.peak_finding import LoadedDataWithFeatures
+from pulse3D.data_loader.metadata import NormalizationMethods
+from pulse3D.pre_analysis import sort_wells_in_df, apply_window_to_df, PreAnalyzedData, post_process
+import structlog
+from utils.s3 import upload_file_to_s3
+
+PULSE3D_UPLOADS_BUCKET = os.getenv("TEST_UPLOADS_BUCKET")
+
+DRY_RUN = True
+
+s3_client = boto3.client("s3")
+
+logging.config.dictConfig(
+    {
+        "version": 1,
+        "disable_existing_loggers": False,
+        "handlers": {
+            "default": {"level": "INFO", "class": "logging.StreamHandler"},
+            "file": {
+                "level": "INFO",
+                "class": "logging.handlers.WatchedFileHandler",
+                "filename": os.path.join(
+                    "logs",
+                    f"extract_metadata__{datetime.datetime.utcnow().strftime('%Y_%m_%d__%H_%M_%S')}.log",
+                ),
+            },
+        },
+        "loggers": {"": {"handlers": ["default", "file"], "level": "INFO", "propagate": True}},
+    }
+)
+structlog.configure(
+    processors=[
+        structlog.processors.StackInfoRenderer(),
+        structlog.dev.set_exc_info,
+        structlog.processors.format_exc_info,
+        structlog.processors.TimeStamper(fmt="%Y-%m-%d %H:%M:%S.%f"),
+        structlog.stdlib.add_log_level,
+        structlog.dev.ConsoleRenderer(colors=False),
+    ],
+    logger_factory=structlog.stdlib.LoggerFactory(),
+    wrapper_class=structlog.stdlib.BoundLogger,
+    cache_logger_on_first_use=True,
+)
+logger = structlog.get_logger()
+
+
+async def main():
+    logger.info("START")
+
+    job_counts = {"total": 0, "success": 0, "failed": 0, "skipped": 0}
+
+    try:
+        DB_PASS = os.getenv("POSTGRES_PASSWORD")
+        DB_USER = os.getenv("POSTGRES_USER", default="curibio_jobs")
+        DB_HOST = os.getenv("POSTGRES_SERVER", default="psql-rds.default")
+        DB_NAME = os.getenv("POSTGRES_DB", default="curibio")
+
+        dsn = f"postgresql://{DB_USER}:{DB_PASS}@{DB_HOST}:5432/{DB_NAME}"
+
+        async with asyncpg.create_pool(dsn=dsn) as pool:
+            async with pool.acquire() as con:
+                all_jobs = await con.fetch(
+                    "select * from jobs_result where meta->>'version' like '1.0%' and meta->>'version' not like '%rc%' and status='finished' order by created_at desc"
+                )
+                job_counts["total"] = len(all_jobs)
+                logger.info(f"found {len(all_jobs)} jobs")
+                for job_info in all_jobs:
+                    res = await process_job(con, job_info)
+                    job_counts[res] += 1
+    finally:
+        logger.info(f"result: {job_counts}")
+        logger.info("DONE")
+
+
+def _create_file_info(
```

Comment:
this is the same function we have in the pulse3d-worker, but I'm not sure it merits a refactor so we can reuse it if we're only running this backfill once

Reply:
Yes, however for the hermes project we do probably want to move this into a shared library at some point